Automattic/wp-calypso
# Jetpack Manage: Remove Owner's User ID field from licenses page as it only shows current user

Related to https://github.com/Automattic/jetpack-manage/issues/113

## Proposed Changes

* This PR removes the 'Owner's User ID' field from the licenses page as it does not serve much purpose - it will only eveer be blank if unassigned, or the current user.

## Testing Instructions

* Check out this PR and navigate to `http://jetpack.cloud.localhost:3000/partner-portal/licenses` in development mode, assuming you have licenses on this page.
* Expand the box for one of the licenses - you should not see any mention of 'Owner's User ID'.

Before:

<img width="1510" alt="Screenshot 2024-01-05 at 10 48 50" src="https://github.com/Automattic/wp-calypso/assets/16754605/9cf59339-048a-4570-b84f-74bd064526b2">


After:

<img width="1500" alt="Screenshot 2024-01-05 at 10 48 58" src="https://github.com/Automattic/wp-calypso/assets/16754605/b589cd58-4d4e-4554-b31b-4d956603475b">


## Pre-merge Checklist
